### PR TITLE
[eclipse/xtext#1777] ported more xtend code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1777] ported more xtend code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/HierarchicalDocumentSymbolService.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+						if (parentSymbol == null) {
+							rootSymbols.add(symbol);
+						} else {
+							parentSymbol.getChildren().add(symbol);
+						}
+					}
+				}
+			}
+		}
+		return rootSymbols.stream().map(symbol -> Either.<SymbolInformation, DocumentSymbol>forRight(symbol))
```
Comment (szarnekow) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/HierarchicalDocumentSymbolService.java:116]: `map(Either::forRight).collection(..)` ?

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/CodeConfig.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+	private String lineDelimiter = Strings.newLine();
+
+	private String fileHeader;
+
+	private String fileHeaderTemplate = "/*\n * generated by Xtext\n */";
+
+	private final List<IClassAnnotation> classAnnotations = new ArrayList<>();
+
+	private boolean preferXtendStubs = true;
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/CodeConfig.java:48]: Sad.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/project/XtextProjectConfig.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		setDefaults();
+		injector.injectMembers(this);
+		for (SubProjectConfig it : getEnabledProjects()) {
+			it.initialize(injector);
+		}
+	}
+
+	public void setDefaults() {
+		runtime.setEnabled(true);
+		if (Lists.newArrayList(eclipsePlugin, web).stream().anyMatch(SubProjectConfig::isEnabled)) {
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/project/XtextProjectConfig.java:82]: `if (eclipsePlugin.isEnabled() || web.isEnabled())`

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/HierarchicalDocumentSymbolService.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/HierarchicalDocumentSymbolService.java:116]: type inference cannnot do that

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/contentassist/ContentAssistService.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+		Position start = null;
+		if (region.getOffset() > doc.getContents().length()) {
+			Position docEnd = doc.getPosition(doc.getContents().length());
+			start = new Position(docEnd.getLine(), docEnd.getCharacter() + region.getLength());
+		} else {
+			start = doc.getPosition(region.getOffset());
+		}
+		Position end = null;
+		if (region.getEndOffset() > doc.getContents().length()) {
+			end = new Position(start.getLine(), start.getCharacter() + region.getLength());
```
[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/contentassist/ContentAssistService.java:195]: am not sure if i want to fix all the bugs

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/contentassist/ContentAssistService.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+		} else {
+			return CompletionItemKind.Value;
+		}
+	}
+
+	protected TextEdit toTextEdit(ReplaceRegion region, Document doc) {
+		Position start = null;
+		if (region.getOffset() > doc.getContents().length()) {
+			Position docEnd = doc.getPosition(doc.getContents().length());
+			start = new Position(docEnd.getLine(), docEnd.getCharacter() + region.getLength());
```
[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/contentassist/ContentAssistService.java:189]: can you please create an separate issue for "highjack my branch" when i am doune with first feedbacxk round

**Pull request merged**